conda/models/environment.py

```diff
@@ -358,3 +425,102 @@ def merge(cls, *environments):
             variables=variables,
             virtual_packages=virtual_packages,
         )
+
+    @classmethod
+    def from_prefix(
+        cls,
+        prefix: str,
+        name: str,
+        platform: str,
+        *,
+        from_history: bool = False,
+        no_builds: bool = False,
+        ignore_channels: bool = False,
+        channels: list[str] | None = None,
+        override_channels: bool = False,
+    ) -> Environment:
+        """
+        Create an Environment model from an existing conda prefix.
+
+        This method analyzes an installed conda environment and creates
+        an Environment model that can be used for exporting or other operations.
+
+        :param prefix: Path to the conda environment prefix
+        :param name: Name for the environment
+        :param platform: Target platform (e.g., 'linux-64', 'osx-64')
+        :param from_history: Use explicit specs from history instead of installed packages
+        :param no_builds: Exclude build strings from package specs
+        :param ignore_channels: Don't include channel information in package specs
+        :param channels: Additional channels to include in environment config
+        :param override_channels: Whether to override default channels
+        :return: Environment model representing the prefix
+        """
+        prefix_data = PrefixData(prefix)
+
+        # Build requested packages from installed packages or history
+        requested_packages = []
+
+        if from_history:
+            # Use explicit specs from history
+            history = History(prefix)
+            spec_map = history.get_requested_specs_map()
+            requested_packages = list(spec_map.values())
+        else:
+            # Read all installed packages from prefix data
+            for prefix_record in prefix_data.iter_records():
+                # Create MatchSpec from installed package
+                if no_builds:
+                    spec_str = f"{prefix_record.name}=={prefix_record.version}"
+                else:
+                    spec_str = f"{prefix_record.name}=={prefix_record.version}={prefix_record.build}"
+
+                if (
+                    not ignore_channels
+                    and prefix_record.channel
+                    and prefix_record.channel.name
+                ):
+                    spec_str = f"{prefix_record.channel.name}::{spec_str}"
+
+                requested_packages.append(MatchSpec(spec_str))
```

Comment:
I think these should be explicit packages instead of requested specs. `requested_packages` represent ones that the user requested, for example the specs that you can get from the history `.get_requested_specs_map`. These packages from the prefix data are all explicit packages.

It's maybe worth it to populate both the `requested_packages` and `explicit_packages` for the env.